go/vt/vttablet/tabletserver/query_engine.go

```diff
@@ -105,6 +105,22 @@ func (ep *TabletPlan) buildAuthorized() {
 	}
 }
 
+// return a sql string with the params stripped out for display
+func redactSQLQuery(sql string) (string, error) {
```

Comment:
I think this function makes more sense to be in the `sqlparser` module.

I added some similar hooks in sqlparser/truncate_query.go so that queries wouldn't be too long for the debug UI in the cases where we have bulk inserts, which we should probably use here too.

So I'd say you should move this into a new `sqlparser.TruncateAndRedactForUI` function inside the sqlparser package.

Reply:
👍 Moved into the sqlparser package. I kept it as just a redact function, since we don't want any truncation here.

Reply:
@jscheinblum I actually disagree and think we do want truncation here. Since all we're getting from the UI is a hint as to which query patterns are being consolidated (given that the bind vars are removed), I think limiting that to the default 512 bytes is consistent with the other debug UIs.